### PR TITLE
feat(mobile): add extended wd commands for appium

### PR DESCRIPTION
**Pull request by sjelin**
Had to make some minor changes to the website to handle longer inheritance
chains

Closes https://github.com/angular/protractor/issues/1940

### CODE REVIEW

File: lib/webdriver-js-extender/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// //  webdriver_extensions.ExtendedWebDriver
+// //
+// /////////////////////////////////////////////////////////////////////////////
+/**
+ * Protractor's `browser` object is a wrapper for an instance of
+ * `ExtendedWebDriver`, provided by `webdriver-js-extender`, which itself is
+ * just an instance of `selenium-webdriver`'s WebDriver with some extra methods
+ * added in. The `browser` object inherits all of WebDriver's and
+ * ExtendedWebDriver's methods, but only the methods most useful to Protractor
+ * users are documented here.
```
Comment (juliemr) [inline lib/webdriver-js-extender/index.js]: This comment should mention that these are available only if your server has appium.

**Comment by juliemr** [comment]
Other than one comment LGTM, and CI is happy, yay!

File: lib/webdriver-js-extender/index.js (hunk 1 above)
Comment (sjelin) [inline lib/webdriver-js-extender/index.js]: Done (see below)

**Comment by sjelin** [comment]
I can't quite check docgen due to #3869, but it looks about as good as it could considering, and before I switch to the beta branch (back in https://github.com/angular/protractor/pull/3326) docgen was good.  I'll merge as soon as I hear back from CI.

File: lib/webdriver-js-extender/index.js (hunk 1 above)
Comment (gkohen) [inline lib/webdriver-js-extender/index.js]: @sjelin , do you see this latest commit as something that can assist testing hybrid app with Protractor?

**Comment by thyagab** [comment]
@juliemr are  selectors to identify elements with accessability id's for native ios app are supported?

**Comment by wswebcreation** [comment]
@thyagab ,

I dont think it's wise to `@` mention individual members. There are a lot contributers that collaborate on the issue PR.

As an answer for your question. [This is the browser support list](https://github.com/angular/protractor/blob/master/docs/browser-support.md). Protractor doesn't (officially) supports Native apps with native selectors.

In a previous project I worked with a Hybrid App with native elements. The app opened with a webview, but some screens were native. To access the native elements with Protractor I used `wd` and `wd-bridge`. I then needed to:

- change the context from `webview` to `native` with for example `wdBrowser.context(context)` where `context` was or `webview` or `native`
- click on an element with for example `wdBrowser.elementById(`something:id/key_cell0`).click()`

Hopefully this helps you

Grtz

Wim

**Comment by thyagab** [comment]
Thank you for response. We are looking for native app support which seems not there at thi spoint.

**Pull request merged**